content/references/protocol-reference/transactions/transaction-types/ammdeposit.md

```diff
@@ -0,0 +1,131 @@
+---
+html: ammdeposit.html
+parent: transaction-types.html
+blurb: Deposit funds into an Automated Market Maker in exchange for LPTokens.
+labels:
+  - AMM
+status: not_enabled
+---
+# AMMDeposit
+[[Source]](https://github.com/gregtatcam/rippled/blob/amm-core-functionality/src/ripple/app/tx/impl/AMMDeposit.cpp "Source")
+<!-- TODO: Update source link to merged version when available -->
+
+{% include '_snippets/amm-disclaimer.md' %}
+
+Deposit funds into an Automated Market-Maker (AMM) instance and receive the AMM's liquidity provider tokens (_LP Tokens_) in exchange. You can deposit one or both of the assets in the AMM's pool.
+
+## Example {{currentpage.name}} JSON
+
+```json
+{
+    "Account" : "rJVUeRqDFNs2xqA7ncVE6ZoAhPUoaJJSQm",
+    "Amount" : {
+        "currency" : "TST",
+        "issuer" : "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd",
+        "value" : "2.5"
+    },
+    "Amount2" : "30000000",
+    "Asset" : {
+        "currency" : "TST",
+        "issuer" : "rP9jPyP5kyvFRb6ZiRghAGw5u8SGAmU4bd"
+    },
+    "Asset2" : {
+        "currency" : "XRP"
+    },
+    "Fee" : "10",
+    "Flags" : 1048576,
+    "Sequence" : 7,
+    "TransactionType" : "AMMDeposit"
+}
+```
+
+{% include '_snippets/tx-fields-intro.md' %}
+
+| Field         | JSON Type           | [Internal Type][] | Required? | Description |
+|:--------------|:--------------------|:------------------|:----------|:------------|
+| `Asset`       | Object              | STIssue           | Yes       | The definition for one of the assets in the AMM's pool. In JSON, this is an object with `currency` and `issuer` fields (omit `issuer` for XRP). |
+| `Asset2`      | Object              | STIssue           | Yes       | The definition for the other asset in the AMM's pool. In JSON, this is an object with `currency` and `issuer` fields (omit `issuer` for XRP). |
+| `Amount`      | [Currency Amount][] | Amount            | No        | The amount of one asset to deposit to the AMM. If present, this must match the type of one of the assets (tokens or XRP) in the AMM's pool. |
+| `Amount2`     | [Currency Amount][] | Amount            | No        | The amount of another asset to add to the AMM. If present, this must match the type of the other asset in the AMM's pool and cannot be the same asset as `Amount`. |
+| `EPrice`      | [Currency Amount][] | Amount            | No        | The maximum effective price, in the deposit asset, to pay for each LP Token received. |
+| `LPTokenOut`  | [Currency Amount][] | Amount            | No        | How many of the AMM's LP Tokens to buy. |
+
+
+### AMMDeposit Modes
+
+This transaction has several modes, depending on which flags you specify. Each mode expects a specific combination of fields. The modes fall into two categories: 
```

Comment:
```suggestion
This transaction has two deposit modes:
```

Reply:
The transaction has 5 modes, two of which are double-asset deposits and three of which are single-asset deposits. Each mode has a specific subset of allowed fields _and_ works differently in terms of how it decides how much money to deposit (where it places the limits). As much as I would like to say it has 2 modes, I think that ends up oversimplifying.

Reply:
I misinterpreted the original meaning, then. Disregard.